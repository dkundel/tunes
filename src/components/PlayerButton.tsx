import * as React from 'react';
import styled from 'styled-components';

export interface PlayerButtonProps {
  onClick?: Function;
  materialIcon: string;
}

const StyledButton = styled.button`
  width: 40px;
  height: 51px;
  border: 0;
  outline: 0;
  cursor: pointer;
  background: var(--colorPrimary);
  color: var(--colorSecondary);
  &:hover {
    background: var(--colorPrimaryHover);
  }
`;

export default class PlayerButton extends React.Component<
  PlayerButtonProps,
  undefined
> {
  public render() {
    return (
      <StyledButton onClick={() => this.props.onClick && this.props.onClick()}>
        <i className="material-icons">
          {this.props.materialIcon || 'help_outline'}
        </i>
      </StyledButton>
    );
  }
}
