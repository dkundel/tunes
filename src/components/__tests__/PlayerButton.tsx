import * as React from 'react';
import PlayerButton from '../PlayerButton';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

describe('test PlayerButton component', () => {
  test('handles materialIcon correctly', () => {
    const component = renderer.create(<PlayerButton materialIcon="search" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('handles empty materialIcon correctly', () => {
    const component = renderer.create(<PlayerButton materialIcon="" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('handles onClick event correct', () => {
    const click = jest.fn();
    const component = shallow(
      <PlayerButton
        materialIcon="search"
        onClick={(...args: any[]) => click(args)}
      />
    );

    expect(component.children().text()).toBe('search');
    component.simulate('click');
    expect(click).toHaveBeenCalledTimes(1);
  });
});
