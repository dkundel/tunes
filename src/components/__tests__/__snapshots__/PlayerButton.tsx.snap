// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PlayerButton component handles empty materialIcon correctly 1`] = `
<button
  className="sc-bdVaJa jGqIse"
  onClick={[Function]}
>
  <i
    className="material-icons"
  >
    help_outline
  </i>
</button>
`;

exports[`test PlayerButton component handles materialIcon correctly 1`] = `
<button
  className="sc-bdVaJa jGqIse"
  onClick={[Function]}
>
  <i
    className="material-icons"
  >
    search
  </i>
</button>
`;
